runtime/ops/os.rs

```diff
@@ -64,13 +64,16 @@ fn op_set_env(
   Ok(json!({}))
 }
 
+#[allow(clippy::unnecessary_wraps)]
 fn op_env(
   state: &mut OpState,
   _args: Value,
   _zero_copy: &mut [ZeroCopyBuf],
 ) -> Result<Value, AnyError> {
-  state.borrow::<Permissions>().env.check()?;
-  let v = env::vars().collect::<HashMap<String, String>>();
+  let perm = &state.borrow::<Permissions>().env;
+  let v = env::vars()
+    .filter(|(var, _)| perm.check(var).is_ok())
```

Comment:
I have a concern whether it does make sense when no env permissions are given at all and Deno.env.toObject() still returning an empty object. I guess in most of the cases calling env.toObject() without any env permission is a mistake (forgot to pass permission). So it might be better to throw in that case perhaps? 🤔

Reply:
I wasn't exactly sure what we would want to do here, so i just implemented this way. i don't mind changing it to being behind full env flag; i just thought this might be a little nice thing to have, though this behaviour might be a tad weird

Reply:
Ah `full env flag` sounds more appropriate here to me. Filtering can be confusing to the users who are very new to Deno and don't know about env flag.